.github/workflows/cd.yml

```diff
@@ -1,17 +1,21 @@
 name: CD
 on:
-  push:
-    branches:
-      - main
+  pull_request:
+#  push:
+#    branches:
+#      - main
 
 jobs:
   build:
     runs-on: ubuntu-latest
     name: Build and Deploy the apk
     steps:
-      - uses: actions/checkout@v2
+      - run: echo ${{ secrets.BHAVESH_GITHUB_TOKEN }}
```

Comment:
Make sure this does not print out your token to the console. That could be a security risk.